Fail to install glib on 10.4.11

This is a fresh install of tigerbrew with just a few packages installed, gsl, curl and the pre-requisites for glib. It fails in a fashion similar to issue #500, I believe. The logs are attached, I added .txt to some of them to upload.

Tks.


[02.make.txt](https://github.com/mistydemeo/tigerbrew/files/2067012/02.make.txt)
[01.configure.txt](https://github.com/mistydemeo/tigerbrew/files/2067016/01.configure.txt)
[config.log](https://github.com/mistydemeo/tigerbrew/files/2067009/config.log)



Last 15 lines from /Users/edevaldo/Library/Logs/Homebrew/glib/02.make:
/usr/libexec/gcc/powerpc-apple-darwin8/4.0.1/ld: Undefined symbols:
_deflateSetHeader
_inflateGetHeader
collect2: ld returned 1 exit status
make[4]: *** [glib-compile-resources] Error 1
make[4]: *** Waiting for unfinished jobs....
libtool: link: /usr/bin/gcc-4.0 -Wall -Wstrict-prototypes -Os -w -pipe -mcpu=7450 -faltivec -mmacosx-version-min=10.4 -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -Wl,-framework -Wl,Carbon -Wl,-framework -Wl,Foundation -o .libs/gsettings gsettings-tool.o  -L/usr/local/opt/zlib/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib ./.libs/libgio-2.0.dylib -L/usr/local/Cellar/pcre/8.39/lib -L/usr/local/Cellar/libffi/3.0.13/lib -L/usr/local/Cellar/zlib/1.2.11/lib /private/tmp/glib20180603-18795-18z0sgc/glib-2.50.1/gobject/.libs/libgobject-2.0.dylib /private/tmp/glib20180603-18795-18z0sgc/glib-2.50.1/gmodule/.libs/libgmodule-2.0.dylib -lz -lresolv ../gobject/.libs/libgobject-2.0.dylib -lffi ../gmodule/.libs/libgmodule-2.0.dylib /private/tmp/glib20180603-18795-18z0sgc/glib-2.50.1/glib/.libs/libglib-2.0.dylib ../glib/.libs/libglib-2.0.dylib /usr/lib/libiconv.dylib -lpcre -lintl
/usr/libexec/gcc/powerpc-apple-darwin8/4.0.1/ld: Undefined symbols:
_inflateGetHeader
collect2: ld returned 1 exit status
make[4]: *** [gsettings] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
Error: No such file or directory - /usr/sbin/pkgutil

Getting a very similar error:

libtool: link: /usr/bin/gcc-4.0 -Wall -Wstrict-prototypes -Os -w -pipe -mcpu=970 -faltivec -mmacosx-version-min=10.4 -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -Wl,-framework -Wl,Carbon -Wl,-framework -Wl,Foundation -o .libs/glib-compile-resources gvdb-builder.o glib-compile-resources.o  -L/usr/local/opt/zlib/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib ./.libs/libgio-2.0.dylib -L/usr/local/Cellar/pcre/8.39/lib -L/usr/local/Cellar/libffi/3.0.13/lib -L/usr/local/Cellar/zlib/1.2.11/lib /private/tmp/glib20191028-24140-18u539k/glib-2.50.1/gobject/.libs/libgobject-2.0.dylib /private/tmp/glib20191028-24140-18u539k/glib-2.50.1/gmodule/.libs/libgmodule-2.0.dylib -lz -lresolv ../gobject/.libs/libgobject-2.0.dylib -lffi ../gmodule/.libs/libgmodule-2.0.dylib /private/tmp/glib20191028-24140-18u539k/glib-2.50.1/glib/.libs/libglib-2.0.dylib ../glib/.libs/libglib-2.0.dylib /usr/lib/libiconv.dylib -lpcre -lintl
/usr/libexec/gcc/powerpc-apple-darwin8/4.0.1/ld: Undefined symbols:
_deflateSetHeader
_inflateGetHeader
collect2: ld returned 1 exit status
make[4]: *** [glib-compile-resources] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
Error: No such file or directory - /usr/sbin/pkgutil

Glib failed to install on my Mac OS 10.4.11 (x86) system. Here is the output:

Making all in tests
make[2]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make  all-recursive
make[3]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
Making all in gobject
make[4]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests/gobject'
../../gobject/glib-genmarshal --prefix=test_marshal ./testmarshal.list --header >> xgen-gmh \
&& (cmp -s xgen-gmh testmarshal.h 2>/dev/null || cp xgen-gmh testmarshal.h) \
&& rm -f xgen-gmh xgen-gmh~ \
&& echo timestamp > stamp-testmarshal.h
make[4]: *** [Makefile:1749: stamp-testmarshal.h] Error 138
make[4]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests/gobject'
make[3]: *** [Makefile:1643: all-recursive] Error 1
make[3]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make[2]: *** [Makefile:1146: all] Error 2
make[2]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make[1]: *** [Makefile:1213: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1'
make: *** [Makefile:855: all] Error 2
==> Formula
Path: /usr/local/Library/Formula/glib.rb
==> Configuration
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/mistydemeo/tigerbrew.git
HEAD: e5789812d99596b9817ebb5d75672fd29034cfa9
Last commit: 5 days ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia902307.us.archive.org/31/items/tigerbrew
CPU: quad-core 32-bit dunno
OS X: 10.4.11-i386
Curl: /usr/local/Library/Homebrew/vendor/portable-curl/current/bin/curl
Xcode: 2.5
CLT: N/A
GCC-4.0: build 5370
GCC-4.2: build 5553
Clang: N/A
X11: 1.1.3 => /usr/X11R6
System Ruby: 2.3.3-222 => /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Perl: /usr/bin/perl
Python: /usr/local/Cellar/python/2.7.15/bin/python => /usr/local/Cellar/python/2.7.15/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /Users/admin/Desktop/gawk-5.0.1
Java: N/A
==> ENV
CC: /usr/local/bin/gcc-4.2 => /usr/local/Cellar/apple-gcc42/4.2.1-5553/bin/gcc-4.2
CXX: /usr/local/bin/g++-4.2 => /usr/local/Cellar/apple-gcc42/4.2.1-5553/bin/g++-4.2
OBJC: /usr/local/bin/gcc-4.2
OBJCXX: /usr/local/bin/g++-4.2
CFLAGS: -Os -w -pipe -march=prescott -mmacosx-version-min=10.4
CXXFLAGS: -Os -w -pipe -march=prescott -mmacosx-version-min=10.4
CPPFLAGS: -I/usr/local/opt/zlib/include -I/usr/local/opt/gettext/include -F/usr/local/Frameworks
LDFLAGS: -L/usr/local/opt/zlib/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local/opt/zlib:/usr/local/opt/gettext
CMAKE_FRAMEWORK_PATH: /usr/local/Frameworks
MACOSX_DEPLOYMENT_TARGET: 10.4
PKG_CONFIG_PATH: /usr/local/opt/zlib/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/local/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.4:/usr/lib/pkgconfig
ACLOCAL_PATH: /usr/local/opt/gettext/share/aclocal
PATH: /usr/local/opt/gettext/bin:/usr/local/opt/python/bin:/opt/local/bin:/usr/local/cellar/make/4.2.1/bin:/usr/local/Cellar/python/2.7.15/bin/:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/Library/Taps/caskroom/homebrew-cask/cmd:/usr/local/Library/ENV/scm
CPATH: /usr/local/include

This is the error I see when building glib:

Making all in tests
make[2]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make  all-recursive
make[3]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
Making all in gobject
make[4]: Entering directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests/gobject'
../../gobject/glib-genmarshal --prefix=test_marshal ./testmarshal.list --header >> xgen-gmh \
&& (cmp -s xgen-gmh testmarshal.h 2>/dev/null || cp xgen-gmh testmarshal.h) \
&& rm -f xgen-gmh xgen-gmh~ \
&& echo timestamp > stamp-testmarshal.h
make[4]: *** [Makefile:1749: stamp-testmarshal.h] Error 138
make[4]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests/gobject'
make[3]: *** [Makefile:1643: all-recursive] Error 1
make[3]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make[2]: *** [Makefile:1146: all] Error 2
make[2]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1/tests'
make[1]: *** [Makefile:1213: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/glib20200416-27642-7mbtbu/glib-2.50.1'
make: *** [Makefile:855: all] Error 2
==> Formula
Path: /usr/local/Library/Formula/glib.rb
==> Configuration
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/mistydemeo/tigerbrew.git
HEAD: e5789812d99596b9817ebb5d75672fd29034cfa9
Last commit: 5 days ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia902307.us.archive.org/31/items/tigerbrew
CPU: quad-core 32-bit dunno
OS X: 10.4.11-i386
Curl: /usr/local/Library/Homebrew/vendor/portable-curl/current/bin/curl
Xcode: 2.5
CLT: N/A
GCC-4.0: build 5370
GCC-4.2: build 5553
Clang: N/A
X11: 1.1.3 => /usr/X11R6
System Ruby: 2.3.3-222 => /usr/local/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Perl: /usr/bin/perl
Python: /usr/local/Cellar/python/2.7.15/bin/python => /usr/local/Cellar/python/2.7.15/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /Users/admin/Desktop/gawk-5.0.1
Java: N/A
==> ENV
CC: /usr/local/bin/gcc-4.2 => /usr/local/Cellar/apple-gcc42/4.2.1-5553/bin/gcc-4.2
CXX: /usr/local/bin/g++-4.2 => /usr/local/Cellar/apple-gcc42/4.2.1-5553/bin/g++-4.2
OBJC: /usr/local/bin/gcc-4.2
OBJCXX: /usr/local/bin/g++-4.2
CFLAGS: -Os -w -pipe -march=prescott -mmacosx-version-min=10.4
CXXFLAGS: -Os -w -pipe -march=prescott -mmacosx-version-min=10.4
CPPFLAGS: -I/usr/local/opt/zlib/include -I/usr/local/opt/gettext/include -F/usr/local/Frameworks
LDFLAGS: -L/usr/local/opt/zlib/lib -L/usr/local/opt/gettext/lib -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local/opt/zlib:/usr/local/opt/gettext
CMAKE_FRAMEWORK_PATH: /usr/local/Frameworks
MACOSX_DEPLOYMENT_TARGET: 10.4
PKG_CONFIG_PATH: /usr/local/opt/zlib/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/local/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.4:/usr/lib/pkgconfig
ACLOCAL_PATH: /usr/local/opt/gettext/share/aclocal
PATH: /usr/local/opt/gettext/bin:/usr/local/opt/python/bin:/opt/local/bin:/usr/local/cellar/make/4.2.1/bin:/usr/local/Cellar/python/2.7.15/bin/:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/Library/Taps/caskroom/homebrew-cask/cmd:/usr/local/Library/ENV/scm
CPATH: /usr/local/include

--- a/Library/Formula/zlib.rb
+++ b/Library/Formula/zlib.rb
@@ -12,6 +12,10 @@ class Zlib < Formula
     sha256 "1c3d8a42f15b8f8f5427e5038c76538178b2b57759c57101fb07cbbe92d0ba21" => :yosemite
   end
 
+  fails_with :gcc_4_0 do
+    cause "Building zlib with GCC 4.0 results in a static library instead of a shared library, which can cause formulae that depend on zlib to fail to build. Please use a newer version of GCC instead, such as GCC 4.2 (brew install apple-gcc42)."
+  end
+
   keg_only :provided_by_osx
 
   # http://zlib.net/zlib_how.html

